src/Core/Content/Newsletter/Event/NewsletterRegisterEvent.php

```diff
@@ -83,9 +83,17 @@ public function getMailStruct(): MailRecipientStruct
             return $this->mailRecipientStruct;
         }
 
+        if ($this->newsletterRecipient->getFirstName() && $this->newsletterRecipient->getLastName()) {
+            return new MailRecipientStruct(
+                [
+                    $this->newsletterRecipient->getEmail() => $this->newsletterRecipient->getFirstName() . ' ' . $this->newsletterRecipient->getLastName(),
+                ]
+            );
+        }
+
```

Comment:
```suggestion
$recipientName = $this->newsletterRecipient->getEmail();

        if ($this->newsletterRecipient->getFirstName() && $this->newsletterRecipient->getLastName()) {
            $recipientName =  $this->newsletterRecipient->getFirstName() . ' ' . $this->newsletterRecipient->getLastName();
            );
        }

```

Reply:
What do you think about this? and using the variable in the last line

Reply:
@shyim you want to send mail to `mail@example.com <mail@example.com>` sounds creapy :-)

Reply:
@shyim thanks, integrated :)
@tinect But that's the plan. The thing is, the mailing service wants to set the recipient properly, but when the only information is the email we thought it better to be that then an empty string.